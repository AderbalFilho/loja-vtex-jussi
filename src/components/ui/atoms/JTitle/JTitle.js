import React from 'react';
import classNames from 'classnames/bind';
import PropTypes from 'prop-types';

import './JTitle.scss';

function JTitle({ children, classes, color, tag, ...rest }) {
  const fontDictionary = {
    h1: 'extra-large',
    h2: 'large',
  };
  const className = classNames(
    'j-title',
    `j-title--${color}`,
    `j-title--${fontDictionary[tag]}`,
    classes
  );
  const HtmlTag = tag;

  return (
    <HtmlTag className={className} {...rest}>
      {children}
    </HtmlTag>
  );
}

JTitle.defaultProps = {
  color: 'black',
};

JTitle.propTypes = {
  children: PropTypes.node,
  classes: PropTypes.string,
  color: PropTypes.oneOf(['black', 'white', 'green']),
  tag: PropTypes.oneOf(['h1', 'h2']).isRequired,
};

export default JTitle;
