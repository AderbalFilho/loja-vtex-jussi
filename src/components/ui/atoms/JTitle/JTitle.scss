@import 'styles/mixin';

.j-title {
  color: var(--black);
}

.j-title--extra-large {
  @extend %text-extra-large;
}

.j-title--large {
  @extend %text-large;
}

.j-title--black {
  color: var(--black);
}

.j-title--white {
  color: var(--white);
}

.j-title--green {
  color: var(--green-jussi);
}
