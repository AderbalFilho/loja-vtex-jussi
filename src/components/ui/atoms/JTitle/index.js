import JTitle from './JTitle';

export default JTitle;
