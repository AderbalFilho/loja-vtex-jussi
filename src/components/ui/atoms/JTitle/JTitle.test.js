import React from 'react';
import TestRenderer from 'react-test-renderer';
import { render } from '@testing-library/react';

import JTitle from './JTitle';

describe('JTitle', () => {
  describe('change tag', () => {
    it('renders h1', () => {
      const { container } = render(<JTitle tag="h1">Jüssi</JTitle>);
      const title = container.querySelector('.j-title');

      expect(title.classList).toContain('j-title--extra-large');
    });

    it('matches snapshot with h1', () => {
      const testRenderer = TestRenderer.create(<JTitle tag="h1">Jüssi</JTitle>);
      const jsonComponent = testRenderer.toJSON();

      expect(jsonComponent).toMatchSnapshot();
    });

    it('renders h2', () => {
      const { container } = render(<JTitle tag="h2">Jüssi</JTitle>);
      const title = container.querySelector('.j-title');

      expect(title.classList).toContain('j-title--large');
    });

    it('matches snapshot with h2', () => {
      const testRenderer = TestRenderer.create(<JTitle tag="h2">Jüssi</JTitle>);
      const jsonComponent = testRenderer.toJSON();

      expect(jsonComponent).toMatchSnapshot();
    });
  });

  describe('change color', () => {
    it('renders black when no color is passed', () => {
      const { container } = render(<JTitle tag="h1">Jüssi</JTitle>);
      const title = container.querySelector('.j-title');

      expect(title.classList).toContain('j-title--black');
    });

    it('renders white', () => {
      const { container } = render(
        <JTitle tag="h1" color="white">
          Jüssi
        </JTitle>
      );
      const title = container.querySelector('.j-title');

      expect(title.classList).toContain('j-title--white');
    });

    it('renders green', () => {
      const { container } = render(
        <JTitle tag="h1" color="green">
          Jüssi
        </JTitle>
      );
      const title = container.querySelector('.j-title');

      expect(title.classList).toContain('j-title--green');
    });

    it('renders black', () => {
      const { container } = render(
        <JTitle tag="h1" color="black">
          Jüssi
        </JTitle>
      );
      const title = container.querySelector('.j-title');

      expect(title.classList).toContain('j-title--black');
    });
  });

  it('renders children', () => {
    const { container } = render(<JTitle tag="h1">Jüssi</JTitle>);
    const title = container.querySelector('.j-title');

    expect(title.textContent).toBe('Jüssi');
  });

  it('renders classes', () => {
    const { container } = render(
      <JTitle tag="h1" classes="title1">
        Jüssi
      </JTitle>
    );
    const title = container.querySelector('.j-title');

    expect(title.classList).toContain('title1');
  });
});
