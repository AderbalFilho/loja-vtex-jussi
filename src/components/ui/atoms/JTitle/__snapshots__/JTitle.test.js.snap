// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JTitle change tag matches snapshot with h1 1`] = `
<h1
  className="j-title j-title--black j-title--extra-large"
>
  Jüssi
</h1>
`;

exports[`JTitle change tag matches snapshot with h2 1`] = `
<h2
  className="j-title j-title--black j-title--large"
>
  Jüssi
</h2>
`;
